runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -351,6 +356,11 @@ class MM_ObjectAllocationAPI
 				/* Do not zero the TLH if it is already zero'd */
 				initializeSlots = initializeSlots && _initializeSlotsOnTLHAllocate;
 #endif /* J9VM_GC_BATCH_CLEAR_TLH */
+#if defined(J9VM_GC_REALTIME)
+				MM_GCRememberedSetFragment *fragment =  &currentThread->sATBBarrierRememberedSetFragment;
+				MM_GCRememberedSet *parent = fragment->fragmentParent;
+				initializeSlots = initializeSlots || (0 != parent->globalFragmentIndex);
```

Comment:
we need a helper (not sure where) and simplify this to something like

```initializeSlots = initializeSlots || isSATBBarrierEnabled(currentThread);```

Reply:
You can try to make it a static helper in  MM_ObjectAccessBarrierAPI (not sure if you'll create an #include cycle, though)
There is a couple of spots in that class (like internalPreStoreObjectSATB) that could use it, too.

Reply:
introduced `isSATBDoubleBarrierEnabled` as well to simply other spots in `MM_ObjectAccessBarrierAPI `